### PR TITLE
Document what provider have support for --cookie-refresh. (Issue #501)

### PR DESCRIPTION
**Pull request by devopsix**
Updates `docs/configuration/configuration.md` as requested in #501.

Before merging please verify:

- Is the list of providers with support for `--cookie-refresh` correct?
- Are footnotes supported by the markdown tool chain used for rendering the documentation?

### CODE REVIEW

**Comment by devopsix** [comment]
Still relevant.

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ An example [oauth2-proxy.cfg]({{ site.gitweb }}/contrib/oauth2-proxy.cfg.example
 | `--cookie-httponly` | bool | set HttpOnly cookie flag | true |
 | `--cookie-name` | string | the name of the cookie that the oauth_proxy creates | `"_oauth2_proxy"` |
 | `--cookie-path` | string | an optional cookie path to force cookies to (ie: `/poc/`) | `"/"` |
-| `--cookie-refresh` | duration | refresh the cookie after this duration; `0` to disable | |
+| `--cookie-refresh` | duration | refresh the cookie after this duration; `0` to disable; not supported by all providers[^2] | |
```
Comment (JoelSpeed) [inline docs/configuration/configuration.md:41]: I think this should be a 1?
```suggestion
| `--cookie-refresh` | duration | refresh the cookie after this duration; `0` to disable; not supported by all providers[^1] | |
```

**Comment by devopsix** [comment]
- You're absolutely right, the first footnote link was incorrect. Change accepted.
- I realized GFM does not support these footnote links. Link syntax changed.
- Rebased onto current master, merge conflicts resolved.

**Comment by JoelSpeed** [comment]
I've spun this up locally and it works, thanks @devopsix

**Pull request merged**